## Задача №1

### Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз. 

Формула Бернулли

In [1]:
from math import factorial, exp
def bernulli(n, k, p):
    comb=factorial(n)/(factorial(k)*factorial(n-k))
    return comb*(p**k)*(1-p)**(n-k)


In [2]:
print(f'P = {bernulli(100,85,0.8): .4f}')

P =  0.0481


#### Ответ:

***P ≈ 0.0481***

## Задача №2

### Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. (a) Какова вероятность, что ни одна из них не перегорит в первый день? (б) Какова вероятность, что перегорят ровно две?

In [3]:
def pois_distr(m,p,n):
    lamb=p*n
    return exp(-lamb)*(lamb**m)/factorial(m)

#### a)

По условию задания: p = 0.0004n = 5000m = 0

In [4]:
print(f'P = {pois_distr(0,0.0004,5000): .4f}')

P =  0.1353


#### б)

По условию задания: p = 0.0004n = 5000m = 2

In [5]:
print(f'P = {pois_distr(2,0.0004,5000): .4f}')

P =  0.2707


#### Ответ:

#### a) ***P ≈ 0.1353***
#### б) ***P ≈ 0.2707***

## Задача №3

### Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

Формула Бернулли

In [6]:
print(f'P = {bernulli(144,70,0.5): .4f}')

P =  0.0628


#### Ответ:

***P ≈ 0.0628***

## Задача №4

### В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. (а) Какова вероятность того, что все мячи белые? (б) Какова вероятность того, что ровно два мяча белые? (в) Какова вероятность того, что хотя бы один мяч белый?

In [7]:
def combs(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [8]:
n1=combs(10,2)
n2=combs(11,2)
print(f'Элементарные исходы:\n n1 = {n1}\n'
f' n2 = {n2}')

Элементарные исходы:
 n1 = 45
 n2 = 55


In [9]:
m1=combs(7,2)
m2=combs(9,2)
print(f'Благоприятные исходы:\n m1 = {combs(7,2)}\n'
      f' m2 = {combs(9,2)}'
     )

Благоприятные исходы:
 m1 = 21
 m2 = 36


#### a)

In [10]:
P=(7/10)*(6/9)*(9/11)*(8/10)
print(f'Вероятность того, что все мячи белые\n Р(а) ≈ {P: .4f}')

Вероятность того, что все мячи белые
 Р(а) ≈  0.3055


#### б)

In [11]:
P1=((7/10)*(6/9)*(2/11)*(1/10))+4*((7/10)*(3/9)*(9/11)*(2/10))+((3/10)*(2/9)*(9/11)*(8/10))
print(f'Вероятность того, что ровно два мяча белые\n Р(б) ≈ {P1: .4f}')

Вероятность того, что ровно два мяча белые
 Р(б) ≈  0.2048


#### в)

In [12]:
P=1-(3/10)*(2/9)*(2/11)*(1/10)
print(f'Вероятность того, что хотя бы один белый мяч\n Р(в) ≈ {P: .4f}')

Вероятность того, что хотя бы один белый мяч
 Р(в) ≈  0.9988


#### Ответ:
***а) Р(а) ≈  0.3055***

***б) Р(б) ≈  0.2048***

***в) Р(в) ≈  0.9988***